File: panel-stage/client/clone-satis-iade-page.js
const fs = require('fs');

const alisPagePath = '/home/azem/projects/otomuhasebe/panel-stage/client/src/app/fatura/alis/page.tsx';
const satisIadePagePath = '/home/azem/projects/otomuhasebe/panel-stage/client/src/app/fatura/iade/satis/page.tsx';

let content = fs.readFileSync(alisPagePath, 'utf8');

// Replacements

// 1. Component name
content = content.replace(/export default function AlisFaturalariPage\(\) \{/, 'export default function SatisIadeFaturalariPage() {');

// 2. Default state API param
content = content.replace(/faturaTipi: 'ALIS'/g, 'faturaTipi: \'SATIS_IADE\'');
// Default form parameter for faturaTipi (if they create via this modal - though usually returns are created from original)
content = content.replace(/faturaTipi: 'SATIS' as 'SATIS' \| 'ALIS'/g, 'faturaTipi: \'SATIS_IADE\' as \'SATIS_IADE\' | \'ALIS_IADE\'');
content = content.replace(/faturaTipi: 'SATIS',(?=\s+cariId)/g, 'faturaTipi: \'SATIS_IADE\',');

// 3. Titles and texts
content = content.replace(/Satın Alma Faturaları/g, 'Satış İade Faturaları');
content = content.replace(/Satın alma faturalarını yönetin/g, 'Satış iade faturalarını görüntüleyin ve yönetin');
content = content.replace(/Yeni Satın Alma Faturası/g, 'Yeni Satış İade Faturası');
content = content.replace(/Yeni Satın Alma İade Faturası/g, ''); // Will remove the return menu item

// Replace path for adding new
content = content.replace(/\/fatura\/alis\/yeni/g, '/fatura/iade/satis/yeni');
content = content.replace(/fatura-alis-yeni/g, 'fatura-iade-satis-yeni');
content = content.replace(/\/fatura\/alis\/duzenle/g, '/fatura/iade/satis/duzenle');
content = content.replace(/\/fatura\/alis\/print/g, '/fatura/iade/satis/print');

// Remove the `İade Oluştur` menu action block since it's already an iade
const returnMenuRegex = /<MenuItem[\s\S]*?key="return"[\s\S]*?onClick=\{\(\) => \{[\s\S]*?const path = `\/fatura\/iade\/alis\/yeni\?originalId=\$\{fatura\.id\}`;[\s\S]*?\}\][\s\S]*?<\/MenuItem>,/g;
content = content.replace(returnMenuRegex, '');


// Fix Colors (Change Amber gradient to Red gradient)
// Amber: linear-gradient(135deg, #f59e0b 0%, #d97706 100%) -> Red: linear-gradient(135deg, #ef4444 0%, #dc2626 100%)
content = content.replace(/135deg, #f59e0b 0%, #d97706 100%/g, '135deg, #ef4444 0%, #dc2626 100%');
content = content.replace(/rgba\(245, 158, 11/g, 'rgba(239, 68, 68');
// Hover Background Amber -> Red Hover
content = content.replace(/135deg, #d97706 0%, #b45309 100%/g, '135deg, #dc2626 0%, #b91c1c 100%');
content = content.replace(/color: '#f59e0b'/g, 'color: \'#ef4444\'');
content = content.replace(/color="#f59e0b"/g, 'color="#ef4444"');

fs.writeFileSync(satisIadePagePath, content);
console.log('Done cloning and replacing!');

File: panel-stage/client/clone-satis-iade-page.py
import re

alis_path = '/home/azem/projects/otomuhasebe/panel-stage/client/src/app/fatura/alis/page.tsx'
satis_iade_path = '/home/azem/projects/otomuhasebe/panel-stage/client/src/app/fatura/iade/satis/page.tsx'

with open(alis_path, 'r', encoding='utf-8') as f:
    content = f.read()

# 1. Component name
content = re.sub(r'export default function AlisFaturalariPage\(\) \{', 'export default function SatisIadeFaturalariPage() {', content)

# 2. Default state API param
content = content.replace("faturaTipi: 'ALIS'", "faturaTipi: 'SATIS_IADE'")
content = content.replace("faturaTipi: 'SATIS' as 'SATIS' | 'ALIS'", "faturaTipi: 'SATIS_IADE' as 'SATIS_IADE' | 'ALIS_IADE'")
content = re.sub(r"faturaTipi: 'SATIS',(?=\s+cariId)", "faturaTipi: 'SATIS_IADE',", content)

# 3. Titles and texts
content = content.replace('Satın Alma Faturaları', 'Satış İade Faturaları')
content = content.replace('Satın alma faturalarını yönetin', 'Satış iade faturalarını görüntüleyin ve yönetin')
content = content.replace('Yeni Satın Alma Faturası', 'Yeni Satış İade Faturası')
content = content.replace('Yeni Satın Alma İade Faturası', '') # Clean return string

# Replace path for adding new
content = content.replace('/fatura/alis/yeni', '/fatura/iade/satis/yeni')
content = content.replace('fatura-alis-yeni', 'fatura-iade-satis-yeni')
content = content.replace('/fatura/alis/duzenle', '/fatura/iade/satis/duzenle')
content = content.replace('/fatura/alis/print', '/fatura/iade/satis/print')

# Replace exact strings for missing content
content = content.replace('Henüz satın alma faturası bulunmamaktadır', 'Henüz satış iade faturası bulunmamaktadır')

# Remove the `İade Oluştur` menu action block
return_menu_regex = r'<MenuItem\s+key="return"[\s\S]*?onClick=\{\(\) => \{[\s\S]*?const path = `/fatura/iade/alis/yeni\?originalId=\$\{fatura\.id\}`;[\s\S]*?\}\][\s\S]*?</MenuItem>,'
content = re.sub(return_menu_regex, '', content)

# Remove `faturaData.aciklama` default in form creation if we want, or just leave it.

# Fix Colors (Change Amber gradient to Red gradient)
content = content.replace('135deg, #f59e0b 0%, #d97706 100%', '135deg, #ef4444 0%, #dc2626 100%')
content = content.replace('rgba(245, 158, 11', 'rgba(239, 68, 68')
content = content.replace('135deg, #d97706 0%, #b45309 100%', '135deg, #dc2626 0%, #b91c1c 100%')
content = content.replace("color: '#f59e0b'", "color: '#ef4444'")
content = content.replace('color="#f59e0b"', 'color="#ef4444"')

# Add specific label change
content = content.replace('Satın Alma Faturası Düzenle', 'Satış İade Faturası Düzenle')
content = content.replace('Yeni Satış Faturası', 'Yeni Satış İade Faturası')
content = content.replace('Satış Faturası Düzenle', 'Satış İade Faturası Düzenle')

with open(satis_iade_path, 'w', encoding='utf-8') as f:
    f.write(content)

print('Done cloning and replacing!')
